core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -330,4 +303,103 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(combinedSchema.schemaProperties());
+    long uid;
+    if (stringId != null) {
+      // If the entity in the store doesn't match the external system, we use the data
+      // of external system to correct it.
+      uid = stringId.id();
+    } else {
+      // If store doesn't exist entity, we sync the entity from the external system.
+      uid = idGenerator.nextId();
+    }
+
+    SchemaEntity schemaEntity =
+        SchemaEntity.builder()
+            .withId(uid)
+            .withName(identifier.name())
+            .withNamespace(identifier.namespace())
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(combinedSchema.auditInfo().creator())
+                    .withCreateTime(combinedSchema.auditInfo().createTime())
+                    .withLastModifier(combinedSchema.auditInfo().lastModifier())
+                    .withLastModifiedTime(combinedSchema.auditInfo().lastModifiedTime())
+                    .build())
+            .build();
+    try {
+      store.put(schemaEntity, true);
+    } catch (Exception e) {
+      LOG.error(FormattedErrorMessages.STORE_OP_FAILURE, "put", identifier, e);
+      throw new RuntimeException("Fail to access underlying storage");
+    }
+
+    return true;
+  }
+
+  private EntityCombinedSchema loadCombinedSchema(NameIdentifier ident) {
+    NameIdentifier catalogIdentifier = getCatalogIdentifier(ident);
+    Schema schema =
+        doWithCatalog(
+            catalogIdentifier,
+            c -> c.doWithSchemaOps(s -> s.loadSchema(ident)),
+            NoSuchSchemaException.class);
+
+    // If the Schema is maintained by the Gravitino's store, we don't have to load again.
+    boolean isManagedSchema = isManagedEntity(catalogIdentifier, Capability.Scope.SCHEMA);
+    if (isManagedSchema) {
+      return EntityCombinedSchema.of(schema)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdentifier,
+                  HasPropertyMetadata::schemaPropertiesMetadata,
+                  schema.properties()))
+          .withImported(true);
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(schema.properties());
+    // Case 1: The schema is not created by Gravitino.
+    if (stringId == null) {
+      return EntityCombinedSchema.of(schema)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdentifier,
+                  HasPropertyMetadata::schemaPropertiesMetadata,
+                  schema.properties()))
+          .withImported(isEntityExist(ident));
+    }
+
+    SchemaEntity schemaEntity =
+        operateOnEntity(
+            ident,
+            identifier -> store.get(identifier, SCHEMA, SchemaEntity.class),
+            "GET",
+            stringId.id());
+
+    boolean imported = schemaEntity != null;
+
+    return EntityCombinedSchema.of(schema, schemaEntity)
```

Comment:
Should you check that two uid are different here?

Reply:
No need. If two uids are different. The `schemaEntity` will be null.